src/app/client/src/app/modules/resource/components/resource/resource.component.ts

```diff
@@ -203,21 +208,26 @@ export class ResourceComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   navigateToCourses(event) {
-    this.router.navigate(['resources/curriculum-course'], {
+    this.router.navigate(['resources/curriculum-courses'], {
       queryParams: {
-        title: _.get(event, 'data.title')
+        title: _.get(event, 'data.title'),
+        board: _.get(this.selectedFilters, 'board'),
+        medium: _.get(this.selectedFilters, 'medium'),
+        gradeLevel: _.get(this.selectedFilters, 'gradeLevel')
       },
     });
   }
 
-  getFilterValues(contents): Array<{title: string, count: number }> {
+  getFilterValues(contents): Array<{title: string, count: string }> {
 
     let subjects = _.map(contents, content => {
       return (_.get(content, 'subject'));
     });
     subjects = _.values(_.groupBy(subjects)).map((subject) => {
-      return ({ title: subject[0], count: subject.length });
-    });
+      return ({ title: subject[0], count: subject.length === 1 ?
+        `${subject.length} ${_.upperCase(this.resourceService.frmelmnts.lbl.sectionTitle1)}`
```

Comment:
can keep the label as **course** and **courses** if it used previously update that with proper key for value used